### PR TITLE
Feat: Support dbt model versioning

### PR DESCRIPTION
**Pull request by izeigerman**
Introduces support for the dbt model versioning feature as described here: https://docs.getdbt.com/docs/collaborate/govern/model-versions

### CODE REVIEW

**Review by tobymao — APPROVED** [review]
what is the consequence of a dbt model version? does it create a different physical table?

**Comment by izeigerman** [comment]
@tobymao for all intents and purposes each version is treated as a separate model. 

This is also consistent with dbt docs:
> How is this different from just creating a new model?
Honestly, it's only a little bit different! There isn't much magic here, and that's by design.

**Comment by sungchun12** [comment]
@tobymao it creates a whole new physical table specific to that version. 


```shell
dbt.exceptions.AmbiguousAliasError: Compilation Error
  dbt found two resources with the database representation "database.schema.model_name".
  dbt cannot create two resources with identical database representations. To fix this,
  change the configuration of one of these resources:
  - model.project_name.model_name.v1 (models/.../model_name.sql)
  - model.project_name.model_name.v2 (models/.../model_name_v2.sql)
```

**Review by georgesittas — APPROVED** [review]

File: sqlmesh/dbt/context.py [hunk 1]
```diff
@@ -185,8 +185,11 @@ def refs(self) -> t.Dict[str, t.Union[ModelConfig, SeedConfig]]:
             for model in t.cast(
                 t.Dict[str, t.Union[ModelConfig, SeedConfig]], {**self._seeds, **self._models}
             ).values():
-                self._refs[model.name] = model
+                self._refs[model.alias or model.name] = model
```
Comment (github-christophe-oudar) [inline sqlmesh/dbt/context.py:188]: I think this change introduce a regression:
`ref` can only be used with the model name AFAIK (I tried to use alias and it fails with dbt).
Moving back to `self._refs[model.name] = model` fixes this issue on my end.

Comment (izeigerman) [reply sqlmesh/dbt/context.py:188]: What issue are you seeing?

**Pull request merged**